sim/midas/src/main/scala/midas/platform/F1Shim.scala

```diff
@@ -8,38 +8,35 @@ import freechips.rocketchip.config.{Parameters, Field}
 import freechips.rocketchip.diplomacy.{LazyModule, LazyModuleImp}
 import freechips.rocketchip.util.HeterogeneousBag
 
-import midas.core.{DMANastiKey}
+import midas.core.{CPUManagedAXI4Key}
 import midas.widgets.{AXI4Printf, CtrlNastiKey}
 import midas.stage.GoldenGateOutputFileAnnotation
 import midas.targetutils.xdc._
 
 case object AXIDebugPrint extends Field[Boolean]
 
-class F1ShimIO(implicit val p: Parameters) extends Bundle {
-  val master = Flipped(new NastiIO()(p alterPartial ({ case NastiKey => p(CtrlNastiKey) })))
-  val dma    = Flipped(new NastiIO()(p alterPartial ({ case NastiKey => p(DMANastiKey) })))
-}
-
 class F1Shim(implicit p: Parameters) extends PlatformShim {
   lazy val module = new LazyModuleImp(this) {
-    val io = IO(new F1ShimIO)
+    val io_master = IO(Flipped(new NastiIO()(p alterPartial { case NastiKey => p(CtrlNastiKey) })))
+    val io_dma    = IO(Flipped(new NastiIO()(p alterPartial {
+      case NastiKey => NastiParameters(p(CPUManagedAXI4Key).get.axi4BundleParams) })))
     val io_slave = IO(HeterogeneousBag(top.module.mem.map(x => x.cloneType)))
 
     if (p(AXIDebugPrint)) {
-      AXI4Printf(io.master, "master")
-      AXI4Printf(io.dma,    "dma")
+      AXI4Printf(io_master, "master")
+      AXI4Printf(io_dma,    "dma")
```

Comment:
Is the plan to later change these names in `aws-fpga`?

Reply:
Yeah. Probably when we pull MMCM generation into the shim.